dags/utility/utility_index_from_s3.py

```diff
@@ -0,0 +1,188 @@
+        f"product_definition_uri:{product_definition_uri} - "
+        f"s3_glob:{s3_glob} - "
+        f"kwargs:{kwargs}"
+    )
+
+    if product_definition_uri and s3_glob:
+        return ADD_PRODUCT_TASK_ID
+    elif not product_definition_uri and s3_glob:
+        return INDEXING_TASK_ID
+    else:
+        raise ValueError('Neither product_definition_uri nor s3_glob was informed!')
+
+
+SET_REFRESH_PRODUCT_TASK_NAME = "parse_dagrun_conf"
+CHECK_DAGRUN_CONFIG = "check_dagrun_config"
+
+
+# THE DAG
+dag = DAG(
+    dag_id=DAG_NAME,
+    doc_md=__doc__,
+    default_args=DEFAULT_ARGS,
+    schedule_interval=None,
+    catchup=False,
+    tags=["k8s", "add-product", "self-service", "index-datasets", "explorer-update"],
+)
+
+with dag:
+    TASK_PLANNER = BranchPythonOperator(
+        task_id=CHECK_DAGRUN_CONFIG,
+        python_callable=check_dagrun_config,
+        op_args=[
+            "{{ dag_run.conf.product_definition_uri }}",
+            "{{ dag_run.conf.s3_glob }}",
+        ],
+    )
+
+    ADD_PRODUCT = KubernetesPodOperator(
+        namespace="processing",
+        image=INDEXER_IMAGE,
+        image_pull_policy="Always",
+        labels={"step": "datacube-product-add"},
+        cmds=["datacube"],
+        arguments=[
+            "product",
+            "add",
+            "{{ dag_run.conf.product_definition_uri }}",
+        ],
+        name="datacube-add-product",
+        task_id=ADD_PRODUCT_TASK_ID,
+        get_logs=True,
+        affinity=ONDEMAND_NODE_AFFINITY,
+        is_delete_operator_pod=True,
+    )
+
+    op_args = [
+        "{{ dag_run.conf.s3_glob }}",
+        "{{ dag_run.conf.products }}",
+        "{{ dag_run.conf.no_sign_request }}",
+        "{{ dag_run.conf.stac }}",
+    ]
+
+    INDEXING = KubernetesPodOperator(
+        namespace="processing",
+        image=INDEXER_IMAGE,
+        image_pull_policy="Always",
+        labels={"step": "s3-to-dc"},
+        cmds=["bash"],
+        arguments=[
+            "-c",
+            "s3-to-dc"
+            "{% if dag_run.conf.stac %}--stac{% endif %} "
+            "{% if dag_run.conf.no_sign_request %}--no-sign-request{% endif %} "
+            "{{ dag_run.conf.s3_glob }} "
+            "{{ dag_run.conf.products }}",
+        ],
+        name='INDEXING_TEST',
+        task_id=INDEXING_TASK_ID,
+        get_logs=True,
+        affinity=ONDEMAND_NODE_AFFINITY,
+        is_delete_operator_pod=True,
+        trigger_rule=TriggerRule.NONE_FAILED_OR_SKIPPED,  # Needed in case add product was skipped
+    )
+
+    # Retrieve product name argument to be sent to Explorer refresh process
+    SET_PRODUCTS = PythonOperator(
+        task_id=SET_REFRESH_PRODUCT_TASK_NAME,
+        python_callable=parse_dagrun_conf,
+        op_args=["{{ dag_run.conf.products }}"],
+    )
+
+    # Execute Explorer Refresh process based on the product name
+    EXPLORER_SUMMARY = explorer_refresh_operator(
+        xcom_task_id=SET_REFRESH_PRODUCT_TASK_NAME,
+    )
+
```

Comment:
what about ows refresh?

Reply:
Not everything is on OWS. I don't think we need to do OWS in this tool.